context/app/static/js/hooks/useSearchData.ts

```diff
@@ -0,0 +1,339 @@
+import { useCallback, useMemo } from 'react';
+import useSWR, { SWRConfiguration } from 'swr';
+import useSWRInfinite, { SWRInfiniteKeyLoader } from 'swr/infinite';
+
+import { fetcher as fetch, multiFetcher as multiFetch } from 'js/helpers/swr';
+import { getAuthHeader, addRestrictionsToQuery } from 'js/helpers/functions';
+import { useAppContext } from 'js/components/Contexts';
+
+import {
+  AggregationsAggregate,
+  SearchHit,
+  SearchRequest,
+  SearchResponseBody,
+} from '@elastic/elasticsearch/lib/api/types';
+import { SWRError } from 'js/helpers/swr/errors';
+
+export interface Hit<Doc extends object | Record<string, unknown>> {
+  _source: Doc;
+}
+
+export interface Hits<Doc extends object | Record<string, unknown>> {
+  searchHits: Hit<Doc>[];
+  isLoading: boolean;
+}
+
+interface UseSearchDataConfig extends SWRConfiguration {
+  useDefaultQuery?: boolean;
+  shouldFetch?: boolean;
+}
+
+interface BuildSearchRequestInitArgs {
+  body: SearchRequest;
+  authHeader: HeadersInit;
+}
+
+/**
+ * Builds the auth header for the request
+ * @returns The auth header if logged in, an empty object otherwise
+ */
+function useAuthHeader() {
+  const { groupsToken } = useAppContext();
+  return useMemo(() => getAuthHeader(groupsToken), [groupsToken]);
+}
+
+/**
+ * Util function to create the request body for the search request
+ * @param arg.query The search request
+ * @param arg.useDefaultQuery Whether to apply the default query restrictions
+ * @returns The request body
+ */
+function createSearchRequestBody({ query, useDefaultQuery }: { query: SearchRequest; useDefaultQuery: boolean }) {
+  return useDefaultQuery ? addRestrictionsToQuery(query) : query;
+}
+
+function buildSearchRequestInit({ body, authHeader }: BuildSearchRequestInitArgs): RequestInit {
+  return {
+    method: 'POST',
+    body: JSON.stringify(body),
+    headers: {
+      'Content-Type': 'application/json',
+      ...authHeader,
+    },
+  };
+}
+
+interface RequestInitArgs {
+  query: SearchRequest;
+  useDefaultQuery: boolean;
+}
+function useRequestInit({ query, useDefaultQuery }: RequestInitArgs) {
+  const authHeader = useAuthHeader();
+  return useMemo(() => {
+    const body = createSearchRequestBody({ query, useDefaultQuery });
+    return buildSearchRequestInit({ body, authHeader });
+  }, [query, useDefaultQuery, authHeader]);
+}
+
+interface UseSearchDataConfig extends SWRConfiguration {
+  useDefaultQuery?: boolean;
+  shouldFetch?: boolean;
+}
+
+interface UseSearchData<Documents, Aggs = Record<string, AggregationsAggregate>> {
+  searchData: SearchResponseBody<Documents, Aggs>;
+  isLoading: boolean;
+}
+
+interface UseHitsData<Documents> {
+  searchHits: NonNullable<Required<SearchHit<Documents>>[]>;
+  isLoading: boolean;
+}
+
+const defaultConfig = {
+  useDefaultQuery: false,
```

Comment:
Looking through the git history, I'm not sure why `useDefaultQuery` was set to false. Perhaps a misnomer, but `useDefaultQuery` extends the es query passed as an arg to avoid docs with `next_revision_uuid` (previous dataset versions) and `sub_status`. We should have this set to true by default. 

Do you remember why you changed it to false here?

Reply:
I'm not sure why it was set to `false`, I'll update this accordingly 👍🏻